src/v0/destinations/braze/util.js

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable camelcase */
+const _ = require('lodash');
+const get = require('get-value');
+const { httpPOST } = require('../../../adapters/network');
+const { processAxiosResponse } = require('../../../adapters/utils/networkUtils');
+const {
+  getDestinationExternalID,
+  getFieldValueFromMessage,
+  removeUndefinedAndNullValues,
+  isDefinedAndNotNull,
+} = require('../../util');
+const { BRAZE_NON_BILLABLE_ATTRIBUTES } = require('./config');
+
+const getEndpointFromConfig = (destination) => {
+  // Init -- mostly for test cases
+  let endpoint = 'https://rest.fra-01.braze.eu';
+
+  // Ref: https://www.braze.com/docs/user_guide/administrative/access_braze/braze_instances
+  if (destination.Config.dataCenter) {
+    const dataCenterArr = destination.Config.dataCenter.trim().split('-');
+    if (dataCenterArr[0].toLowerCase() === 'eu') {
+      endpoint = `https://rest.fra-${dataCenterArr[1]}.braze.eu`;
+    } else {
+      endpoint = `https://rest.iad-${dataCenterArr[1]}.braze.com`;
+    }
+  }
+  return endpoint;
+};
+
+const BrazeDedupUtility = {
+  async doLookup(inputs) {
+    const externalIds = [];
+    const aliasIds = [];
+    // eslint-disable-next-line no-restricted-syntax
+    for (const input of inputs) {
+      const { message } = input;
+      const brazeExternalId = getDestinationExternalID(message, 'brazeExternalId');
+      const userId = getFieldValueFromMessage(message, 'userIdOnly');
+      const anonymousId = get(message, 'anonymousId');
+      if (brazeExternalId) {
+        externalIds.push(brazeExternalId);
+      } else if (userId) {
+        externalIds.push(userId);
+      } else if (anonymousId) {
+        aliasIds.push(anonymousId);
+      }
+    }
+    const externalIdsToQuery = [...new Set(externalIds)];
+    const aliasIdsToQuery = [...new Set(aliasIds)];
+
+    const identifiers = [];
+    if (externalIdsToQuery.length > 0) {
+      externalIdsToQuery.forEach((externalId) => {
+        identifiers.push({
+          external_id: externalId,
+        });
+      });
+    }
+    if (aliasIdsToQuery.length > 0) {
+      aliasIdsToQuery.forEach((aliasId) => {
+        identifiers.push({
+          alias_name: aliasId,
+          alias_label: 'rudder_id',
+        });
+      });
+    }
+    const identfierChunks = _.chunk(identifiers, 50);
+
+    const chunkedUserData = await Promise.all(
```

Comment:
It looks like if any call fails, the entire process will fail right?

Reply:
No it will be handled by `processAxiosResponse` and for that call the users will be an array of `undefined` values